spec/locale.html

```diff
@@ -368,6 +368,20 @@ <h1>Properties of Intl.Locale Instances</h1>
   <emu-clause id="sec-intl-locale-abstracts">
     <h1>Abstract Operations for Locale Objects</h1>
 
+    <emu-clause id="sec-getlocalebasename" type="abstract operation">
+      <h1>
+        GetLocaleBaseName (
+          _locale_: a String,
+        ): a String
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Assert: _locale_ can be matched by the <code>unicode_locale_id</code> Unicode locale nonterminal.
```

Comment:
Can this Assert be moved into the header?

Reply:
Is there any precedent for this kind of type declaration? Do you have something like this in mind?

```
GetLocaleBaseName (
  _locale_: a String matching `unicode_locale_id`,
): a String matching `unicode_language_id`
```

Unless we're already using something similar elsewhere, I'd prefer to not introduce new kinds of type declarations at this point of time.